File: #204_Count_Primes/Solution.java
/* This naive solution exceeds time limit */

public class Solution {
    public int countPrimes(int n) {
    	/* a non-negative number n */
    	if (n < 0) throw new IllegalArgumentException();
        if (n <= 1) return 0;
        int count = 0;
        for (int i = 0; i < n; i++) {
        	if (isPrime(i)) count++;
        }
        return count;
    }
    /* judge if a number is prime */
    public boolean isPrime(int p) {
    	if (p == 0) return false;
    	if (p == 1 || p == 2) return true;
    	if (p % 2 == 0) return false;
    	for (int i = 2; i < p; i++) {
    		if (p % i == 0) return false;
    	}
    	return true;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	System.out.println(sol.isPrime(11));
    }
}

File: #204_Count_Primes/Solution.py
'''
time: O(n * n ^0.5)
'''

class Solution(object):
    def countPrimes(self, n):
    	count = 0
    	for x in range(2, n):
    		if self.isPrime(x):
    			count += 1
    	return count

    def isPrime(self, n):
    	if n <= 1:
    		return False
    	i = 2
    	while i * i <= n:
    		if n % i == 0:
    			return False
    		i += 1
    	return True

sol = Solution()
print sol.countPrimes(3)